libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1701,7 +1701,8 @@ bool PbftManager::pushCertVotedPbftBlockIntoChain_(taraxa::blk_hash_t const &cer
     return false;
   }
 
-  if (!pushPbftBlock_(std::move(period_data_), std::move(current_round_cert_votes), std::move(dag_blocks_order))) {
+  if (!pushPbftBlock_(std::move(period_data_), std::move(current_round_cert_votes), std::move(dag_blocks_order),
+                      true)) {
```

Comment:
After this change we are always waiting for block execution before moving to next period, right? It is not good because we are adding some pause to consensus, while we applying block. Even worse, the block execution time could differ from node to node. With this we are desyncronize the nodes

Reply:
true, I looked at it from the perspective when there is no delegation_delay, in which case you simply have to wait for finalization and you can wait either here or in updateDpos...